infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/server/RouteServerExposer.java

```diff
@@ -124,6 +128,10 @@ public void expose(
             .withTargetPort(servicePort.getName())
             .withServers(externalServers)
             .withLabels(labels)
+            .withHost(
+                domainSuffix != null
+                    ? NameGenerator.generate("route", "." + domainSuffix, 10)
```

Comment:
maybe would looks nicer to separate somehow the `route` and random part? `route-<random>.<suffix>`? Maybe does not make sense, just an idea.

Reply:
I would save one character for the users' domain. Anyway, this router is something that we don't expect that the user will manually have to work with.